src/definitions.ts

```diff
@@ -5,6 +5,14 @@ export interface GenericOAuth2Plugin {
    * @returns {Promise<any>} the resource url response
    */
   authenticate(options: OAuth2AuthenticateOptions): Promise<any>;
+  /**
+   * Handle OAuth implicit flow
+   * @param {OAuth2RedirectAuthenticationOptions} options
+   * @returns {Promise<any>} the token endpoint response
+   */
+  handleRedirectAuthentication(
```

Comment:
Isn't it possible to integration the implicit flow into the existing authenticate method. So this new flow is simply supported by configuration rather than having a new method?

Reply:
@moberwasserlechner , with the implicit flow, after a successful authentication (or cancelling/going back), the app is loaded again; so there's no way to "wait" the `authenticate` call result.

I could add the code in the `authenticate` call, but you would still need to call `authenticate` (right now calling `handleRedirectAuthentication`) on the app startup to successfully read the query string parameters of the `code` and generate an access token correctly.

If you have any other idea to implement this, I could work on that.

Reply:
@eduardoRoth  I did not think the flow through before but your right a handler method is needed but please make sure that it is not mistakenly used instead of authenticate. Either by docs or an alternative method name that makes it more clear. Unfortunately I'm no help with coming up with good names ;)